runelite-client/src/test/java/net/runelite/client/plugins/xpglobes/XpGlobesTest.java

```diff
@@ -0,0 +1,167 @@
+
+		runFirstStatChangeForCaching(virtualLevelTotalXp, Skill.AGILITY);
+		assertTrue(xpGlobesPlugin.getXpGlobes().isEmpty());
+	}
+
+	@Test
+	public void testVirtualLevelInGlobeIsNotShownByDefault()
+	{
+		StatChanged secondStatChangedEvent = new StatChanged(
+			Skill.AGILITY,
+			virtualLevelTotalXp + 1,
+			99,
+			0
+		);
+
+		// The next xp change adds a globe
+		xpGlobesPlugin.onStatChanged(secondStatChangedEvent);
+		assertEquals(Experience.MAX_REAL_LEVEL, xpGlobesPlugin.getXpGlobes().get(0).getCurrentLevel());
+	}
+
+	@Test
+	public void testVirtualLevelInGlobeIsShownWhenConfigured()
+	{
+		when(xpGlobesConfig.showVirtualLevel()).thenReturn(true);
+
+		StatChanged secondStatChangedEvent = new StatChanged(
+			Skill.AGILITY,
+			virtualLevelTotalXp + 1,
+			99,
+			0
+		);
+
+		// The next xp change adds a globe
+		xpGlobesPlugin.onStatChanged(secondStatChangedEvent);
+		assertEquals(Experience.getLevelForXp(virtualLevelTotalXp + 1), xpGlobesPlugin.getXpGlobes().get(0).getCurrentLevel());
+	}
+
+	@Test
+	public void testGlobeIsNotShownWhenHideMaxAndShowVirtualLevelConfigured()
+	{
+		when(xpGlobesConfig.hideMaxed()).thenReturn(true);
+
+		StatChanged secondStatChangedEvent = new StatChanged(
+			Skill.AGILITY,
+			virtualLevelTotalXp + 1,
+			99,
+			0
+		);
+
+		// The next xp change adds a globe
+		xpGlobesPlugin.onStatChanged(secondStatChangedEvent);
+		assertEquals(Collections.emptyList(), xpGlobesPlugin.getXpGlobes());
+	}
+
+	@Test
+	public void testGlobeIsNotShownWhenHideMaxConfigured()
+	{
+		when(xpGlobesConfig.hideMaxed()).thenReturn(true);
+
+		StatChanged secondStatChangedEvent = new StatChanged(
+			Skill.AGILITY,
+			virtualLevelTotalXp + 1,
+			99,
+			0
+		);
+
+		// The next xp change adds a globe
+		xpGlobesPlugin.onStatChanged(secondStatChangedEvent);
+		assertEquals(Collections.emptyList(), xpGlobesPlugin.getXpGlobes());
+	}
+
+	@Test
+	public void testGlobeIsShownOnXpGainBelowMaxWhenHideMaxConfigured()
+	{
+		int totalXp = 1;
+		runFirstStatChangeForCaching(totalXp, Skill.FARMING);
+		assertTrue(xpGlobesPlugin.getXpGlobes().isEmpty());
+
+		StatChanged secondStatChangedEvent = new StatChanged(
+			Skill.FARMING,
+			totalXp + 150,
+			Experience.getLevelForXp(totalXp + 150),
+			0
+		);
+
+		// The next xp change adds a globe
+		xpGlobesPlugin.onStatChanged(secondStatChangedEvent);
+		assertEquals(Experience.getLevelForXp(totalXp + 150), xpGlobesPlugin.getXpGlobes().get(0).getCurrentLevel());
+	}
+
+	@Test
+	public void testStatChangesFromBoostDoNotAffectXpGlobes()
+	{
+		StatChanged boostEvent = new StatChanged(
+			Skill.AGILITY,
+			virtualLevelTotalXp,
+			Experience.getLevelForXp(virtualLevelTotalXp),
+			5
+		);
+
+		// XP hasn't changed since the first event, so no globe should be added
+		xpGlobesPlugin.onStatChanged(boostEvent);
+		assertTrue(xpGlobesPlugin.getXpGlobes().isEmpty());
+	}
+
+	void runFirstStatChangeForCaching(int totalXp, Skill skill)
```

Comment:
Whoops I missed this, this should be private.

Also, the tests in general can be changed to use this to create and dispatch StatChanged events. e.g.
```java
@Test
public void testVirtualLevelInGlobeIsNotShownByDefault()
{
	statChanged(Skill.AGILITY, Experience.getXpForLevel(99) + 1);
	assertEquals(Experience.MAX_REAL_LEVEL, xpGlobesPlugin.getXpGlobes().get(0).getCurrentLevel());`
}
```

The only thing you'd need then is to add a `boostedLevel` overload to this method for the `testStatChangesFromBoostDoNotAffectXpGlobes()` test.
```suggestion
	private void statChanged(Skill skill, int totalXp)
```

Reply:
Method made private, all tests now use the `statChanged` method, except the boost test, which utilizes the overloaded version.